signer/src/transaction_coordinator.rs

```diff
@@ -2606,14 +2603,33 @@ pub async fn should_coordinate_dkg(
 
 /// Assert, given the last dkg and smart contract current aggregate key, if we
 /// need to verify the shares and/or issue a rotate key call.
-pub fn assert_rotate_key_action(
+pub async fn assert_rotate_key_action<C>(
+    context: &C,
     last_dkg: &model::EncryptedDkgShares,
     current_aggregate_key: Option<PublicKey>,
-) -> Result<(bool, bool), Error> {
+) -> Result<(bool, bool), Error>
+where
+    C: Context,
+{
     let needs_rotate_key = Some(last_dkg.aggregate_key) != current_aggregate_key;
 
+    // Check if past verification window, if we are, skip verification
+    let dkg_verification_window = context.config().signer.dkg_verification_window;
+    let max_verification_height = last_dkg
+        .started_at_bitcoin_block_height
+        .saturating_add(dkg_verification_window as u64);
+
+    // Get the current bitcoin chain tip to check against the verification window
+    let current_bitcoin_chain_tip = context
+        .get_storage()
+        .get_bitcoin_canonical_chain_tip_ref()
+        .await?
+        .ok_or(Error::NoChainTip)?;
+
+    let past_verification_window = current_bitcoin_chain_tip.block_height > max_verification_height;
+
     let needs_verification = match last_dkg.dkg_shares_status {
-        model::DkgSharesStatus::Unverified => true,
+        model::DkgSharesStatus::Unverified => !past_verification_window,
         model::DkgSharesStatus::Verified => needs_rotate_key,
```

Comment:
This is the state where we want to avoid verifying if `past_verification_window`